### PR TITLE
Feature/vulnerability audit

### PR DESCRIPTION
**Pull request by rav-pradhan**
### What

Added vulnerability audit to the CI build with accordingly updated Makefile. There were two tickets for this work which I've condensed into this PR due to their changes aligning.

The project has also been updated to use Go modules.

- Added a Go package audit task (`nancy`); [ticket here](https://trello.com/c/BfIrLcv5/83-add-dependency-auditing-to-dp-developer-site-1-2)
- Added `npm audit` command for reviewing js dependencies; [ticket here](https://trello.com/c/XjywGDNM/77-add-js-dependency-auditing-to-developer-site-1-2)

### How to review

- Check to see that the updates to the scripts make sense. This isn't a domain I've worked in before to any great deal so any feedback would be helpful

### Who can review

Anyone but me

### CODE REVIEW

File: ci/audit.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+image_resource:
+  type: docker-image
+  source:
+    repository: onsdigital/dp-concourse-tools-node-go
+    tag: latest
+
+inputs:
+  - name: dp-developer-site
+    path: go/src/github.com/ONSdigital/dp-developer-site
```
Comment (janderson2) [inline ci/audit.yml:13]: You can remove the path now that this repo has be converted to modules. You will need to do the same for the `build.yml` and `unit.yml`. See [this guide on converting to modules](https://github.com/ONSdigital/dp/blob/master/guides/MODULES.md) for more information.

File: ci/audit.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+outputs:
+  - name: build
+
+run:
+  path: go/src/github.com/ONSdigital/dp-developer-site/ci/scripts/audit.sh
```
Comment (janderson2) [inline ci/audit.yml:19]: See comment above about the path.

File: ci/scripts/audit.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash -eux
+
+export cwd=$(pwd)
+
+pushd $cwd/dp-developer-site
```
Comment (janderson2) [inline ci/scripts/audit.sh:5]: This path is correct, but you will need to update this line in the `build.sh` to match after moving to go modules.

File: ci/audit.yml (hunk 1 above)
Comment (rav-pradhan) [reply ci/audit.yml]: Have removed the inputs.path

File: ci/audit.yml (hunk 2 above)
Comment (rav-pradhan) [reply ci/audit.yml]: Have updated this as per the modules guide and removed the go route prefix from the path

File: ci/scripts/audit.sh (hunk 3 above)
Comment (rav-pradhan) [reply ci/scripts/audit.sh:5]: Updated

File: ci/scripts/build.sh [hunk 4]
```diff
@@ -4,8 +4,8 @@ cwd=$(pwd)
 
 export GOPATH=$cwd/go
```
Comment (janderson2) [inline ci/scripts/build.sh:5]: This variable won't be used anymore, so it is probably best to remove it for cleanliness.

File: ci/scripts/audit.sh [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash -eux
```
Comment (janderson2) [inline ci/scripts/audit.sh:1]: It looks like we are missing execute permissions on this file:

```
chmod 755 ci/scripts/audit.sh
```
Should sort things out.

Comment (rav-pradhan) [reply ci/scripts/audit.sh:1]: Have updated

**Pull request merged**